var/spack/repos/builtin/packages/flux-core/package.py

```diff
@@ -0,0 +1,144 @@
+    depends_on("jansson")
+    depends_on("yaml-cpp")
+
+    # versions up to 0.8.0 uses pylint to check Flux's python binding
+    # later versions provide a configure flag and disable the check by default
+    depends_on("py-pylint", when='@:0.8.0')
+
+    depends_on("asciidoc", type='build', when="+docs")
+
+    # Need autotools when building on master:
+    depends_on("autoconf", type='build', when='@master')
+    depends_on("automake", type='build', when='@master')
+    depends_on("libtool", type='build', when='@master')
+
+    def setup(self):
+        pass
+
+    @when('@master')
+    def setup(self):
+        # Allow git-describe to get last tag so flux-version works:
+        git = which('git')
+        git('fetch', '--unshallow')
+
+    def autoreconf(self, spec, prefix):
+        self.setup()
+        if not os.path.exists('configure'):
+            # Bootstrap with autotools
+            bash = which('bash')
+            bash('./autogen.sh')
+
+    @property
+    def lua_version(self):
+        return self.spec['lua'].version.up_to(2)
+
+    @property
+    def lua_share_dir(self):
+        return os.path.join('share', 'lua', str(self.lua_version))
+
+    @property
+    def lua_lib_dir(self):
+        return os.path.join('lib', 'lua', str(self.lua_version))
+
+    def setup_environment(self, spack_env, run_env):
+        #  Ensure ./fluxometer.lua can be found during flux's make check
+        spack_env.append_path('LUA_PATH', './?.lua', separator=';')
+
+        run_env.prepend_path(
+            'LUA_PATH',
+            os.path.join(self.spec.prefix, self.lua_share_dir, '?.lua'),
+            separator=';')
+        run_env.prepend_path(
+            'LUA_CPATH',
+            os.path.join(self.spec.prefix, self.lua_lib_dir, '?.so'),
+            separator=';')
+        run_env.prepend_path(
+            'PYTHONPATH',
+            os.path.join(
+                self.spec.prefix.lib,
+                "python{}".format(self.spec['python'].version.up_to(2)),
+                "site-packages"),
+        )
+        run_env.prepend_path(
+            'FLUX_MODULE_PATH',
+            os.path.join(self.spec.prefix.lib, 'flux', 'modules'))
+        run_env.prepend_path(
+            'FLUX_MODULE_PATH',
+            os.path.join(self.spec.prefix.lib, 'flux', 'modules', 'sched'))
+        run_env.prepend_path(
+            'FLUX_EXEC_PATH',
+            os.path.join(self.spec.prefix.libexec, 'flux', 'cmd'))
+        run_env.prepend_path(
+            'FLUX_RC_PATH',
+            os.path.join(self.spec.prefix, 'etc', 'flux'))
+
+    def setup_dependent_environment(self, spack_env, run_env, dependent_spec):
+        self.setup_environment(spack_env, run_env)
+
+    def configure_args(self):
+        args = ['--enable-pylint=no']
+        if '+docs' not in self.spec:
+            args.append('--disable-docs')
+        return args
+
+    # Default AutotoolsPackage check method fails to find 'check' target
```

Comment:
Let me look into this. I overhauled this method quite a bit in #8223. Do you still have this problem on the latest develop?

Reply:
Based on that output, I think I know what the problem is. The first concerning thing is that `make -q check` shouldn't run `make check`. From the man page:
```
       -q, --question
            ``Question  mode''.   Do  not run any commands, or print anything;
            just return an exit status that is zero if the  specified  targets
            are already up to date, nonzero otherwise.
```
This leads me to believe that you aren't using GNU Make. Can you give me the output of:
```console
$ make --version
```
As for why it isn't being added to `spack-build.out`, I think that's because it's an internal check, not part of the build phase. The actual call to `make check` (without the `-q`) _should_ be in `spack-build.out`.

Reply:
Yeah, something is definitely busted. No idea why `make check` would work but `make -q check` would error out. We may have to ignore the error code and instead look for:
```
make: *** No rule to make target `test'.  Stop.
```
as an indicator of whether or not the target exists.

Reply:
For the purposes of this PR, would you like me to keep or remove the `check` method from the `flux-core` and `flux-sched` packages?

Reply:
Alternatively, we might be able to check the exit status of `make -n`.

Reply:
So taking another look at the documentation for `-q`:
```
      -q, --question
            ``Question mode''.  Do not run any commands, or print anything; just return an exit status that is zero if the specified targets are already up to date, nonzero otherwise.
```
It only returns an exit status of zero if the specified target is up to date.  This will never be the case with flux-sched immediately after running `make`, because there are test-specific files that are only compiled when `make check` is called.  So I don't think the `--question` flag is broken, I think it is just asking a different question.

Reply:
> Alternatively, we might be able to check the exit status of `make -n check`

I agree, `-n` seems to be the way to go:
```
# herbein1 at hype2 in ~/Repositories/flux-framework/flux-sched/sched on git:upmaster ✖︎ [20:12:30]
→ make -n check > /dev/null && echo "Success" || echo "Failure"
Success
# herbein1 at hype2 in ~/Repositories/flux-framework/flux-sched/sched on git:upmaster ✖︎ [20:12:37]
→ make -n foobadfansdfionj > /dev/null && echo "Success" || echo "Failure"
make: *** No rule to make target `foobadfansdfionj'.  Stop.
Failure
```